### DISCUSSION TITLE
根据schema里的rules定义，校验表单的值是否符合要求

### DISCUSSION THREAD

**Discussion by bobby-ai** [alibaba/formily#2052] [Q&A 2.x] [answered]
目前官方给出的例子基于定义的schema生成表单，然后在表单submit时调用form实例内部的validate方法校验field是否符合rules要求。我的问题是：
schema并没有渲染生成Form，同时已有表单的值values，请问2.0版本内部有没有相关方法可以校验values是否满足schema里的rules定义？举个例子：
`
schema: {
    type: 'object',
    properties: {
          feature: {
            type: 'string',
            title: '拆分方式',
            required: true,
            default: '按比例拆分',
            'required': true
            'x-decorator': 'FormItem',
            'x-component': 'Select',
            enum: ['ratio1', 'ratio2'],
            'x-component-props': {
              placeholder: '选择字段'
            },
         }
}
`
对应表单的值values假设 = {feature: ''}，由于feature没有值，所以根据schema定义，此时校验时应该要报错。

**Comment by janryWang** [comment, marked as answer]
参考 https://github.com/alibaba/formily/discussions/1797

**Reply by bobby-ai** [reply]
多谢，真是回复神速，👍
